### PR TITLE
Updates from track-based muon alignment group

### PR DESCRIPTION
**Pull request by aysent**
This pull request includes updates from the track-based muon alignment group that were developed during LS1 and throughout year 2015:

https://github.com/cms-mual
https://github.com/cms-mual/Alignment

The code has been used to deliver a new alignment of the muon system for Run2 in August and November 2015.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
please test

**Comment by aysent** [comment]
@diguida, @cerminar, @franzoni, @mmusich, @davidlange6 please let me know if you have any comments or questions regarding these updates, which would help to continue review of this pull request.

**Comment by mmusich** [comment]
@aysent thanks for the ping. We are reviewing your changes.

**Comment by mmusich** [comment]
@aysent this is going to conflict when https://github.com/cms-sw/cmssw/pull/12995 is merged. Since that one is mature enough to go, you will need to rebase yours once that one is merged.

**Comment by mmusich** [comment]
please test

**Comment by aysent** [comment]
@mmusich #12995 was merged into CMSSW_8_0_X_2016-01-29-1100. Could you please start tests one more time? Thank you in advance.

**Comment by mmusich** [comment]
please test

File: Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py [hunk 1]
```diff
@@ -102,6 +102,30 @@
...
+
+process.MuonNumberingInitialization = cms.ESProducer("MuonNumberingInitialization")
+process.MuonNumberingRecord = cms.ESSource( "EmptyESSource",
+    recordName = cms.string( "MuonNumberingRecord" ),
+    iovIsRunNotTime = cms.bool( True ),
+    firstValid = cms.vuint32( 1 )
+)
+
+process.load("Configuration.StandardSequences.GeometryDB_cff")
+process.load('Configuration.StandardSequences.MagneticField_38T_PostLS1_cff')
```
Comment (mmusich) [inline Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py]: @aysent why not to use the standar include here:
```
process.load('Configuration.StandardSequences.MagneticField_AutoFromDBCurrent_cff') 
```

File: Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py [hunk 2]
```diff
@@ -206,7 +230,7 @@
     process.looper.monitorConfig.AlignmentMonitorMuonVsCurvature.doDT = doDT
     process.looper.monitorConfig.AlignmentMonitorMuonVsCurvature.doCSC = doCSC
 
-process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_condDBv2_cff")
 process.GlobalTag.globaltag = cms.string(globaltag)
```
Comment (mmusich) [inline Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py]: this is not needed anymore in 80X, please revert to the previous load statement

File: Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py (hunk 1 above)
Comment (aysent) [inline Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py]: @mmusich thanks for suggestion. The content of "MagneticField_AutoFromDBCurrent_cff" says that it is obsolete and will be deprecated at some point, and suggests ti replace it with "MagneticField_cff".

File: Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py (hunk 2 above)
Comment (aysent) [inline Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py]: @mmusich done.

**Comment by mmusich** [comment]
please test

File: Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py (hunk 1 above)
Comment (mmusich) [inline Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py]: @aysent yes, thank you

**Comment by mmusich** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**